src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ParenPadCheck.java

```diff
@@ -253,31 +267,10 @@ private void processExpression(DetailAST ast) {
             else if (currentNode.getType() == TokenTypes.RPAREN && !isInTypecast(currentNode)) {
                 processRight(currentNode);
             }
-            else if (currentNode.hasChildren() && !isAcceptableToken(currentNode)) {
-                // Traverse all subtree tokens which will never be configured
-                // to be launched in visitToken()
-                currentNode = currentNode.getFirstChild();
-                continue;
-            }
-
-            // Go up after processing the last child
-            while (currentNode.getNextSibling() == null && currentNode.getParent() != ast) {
-                currentNode = currentNode.getParent();
-            }
             currentNode = currentNode.getNextSibling();
         }
```

Comment:
removed the deep scan and just check the first child and all of its siblings

```
PS D:\test> cat src/Test3.java

public class Test3 {
    int i =  (( (5*4) + ( 4 + 2)));
}
PS D:\test> java  -jar checkstyle-10.14.2-all.jar -t  src/Test3.java
COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
`--CLASS_DEF -> CLASS_DEF [2:0]
    |--MODIFIERS -> MODIFIERS [2:0]
    |   `--LITERAL_PUBLIC -> public [2:0]
    |--LITERAL_CLASS -> class [2:7]
    |--IDENT -> Test3 [2:13]
    `--OBJBLOCK -> OBJBLOCK [2:19]
        |--LCURLY -> { [2:19]
        |--VARIABLE_DEF -> VARIABLE_DEF [3:4]
        |   |--MODIFIERS -> MODIFIERS [3:4]
        |   |--TYPE -> TYPE [3:4]
        |   |   `--LITERAL_INT -> int [3:4]
        |   |--IDENT -> i [3:8]
        |   |--ASSIGN -> = [3:10]
        |   |   `--EXPR -> EXPR [3:13]
        |   |       |--LPAREN -> ( [3:13]         <-- this and all its siblings only checked 
        |   |       |--LPAREN -> ( [3:14]
        |   |       |--PLUS -> + [3:22]
        |   |       |   |--LPAREN -> ( [3:16]          <-- no deep scan so this will not be picked
        |   |       |   |--STAR -> * [3:18]
        |   |       |   |   |--NUM_INT -> 5 [3:17]
        |   |       |   |   `--NUM_INT -> 4 [3:19]
        |   |       |   |--RPAREN -> ) [3:20]
        |   |       |   |--LPAREN -> ( [3:24]
        |   |       |   |--PLUS -> + [3:28]
        |   |       |   |   |--NUM_INT -> 4 [3:26]
        |   |       |   |   `--NUM_INT -> 2 [3:30]
        |   |       |   `--RPAREN -> ) [3:31]
        |   |       |--RPAREN -> ) [3:32]
        |   |       `--RPAREN -> ) [3:33]
        |   `--SEMI -> ; [3:34]
        `--RCURLY -> } [4:0]

```